src/features/class-query-block-context.php

```diff
@@ -20,99 +20,127 @@
 use Alley\WP\WP_Curate\Must_Include_Curated_Posts;
 use Alley\WP\WP_Curate\Plugin_Curated_Posts;
 use Alley\WP\WP_Curate\Recorded_Curated_Posts;
+use Alley\WP\WP_Curate\Trending_Post_Queries;
 use WP_Block;
 use WP_Block_Type;
 use WP_Block_Type_Registry;
 
 /**
  * Provides context to query blocks
  */
-final class Query_Block_Context implements Feature {
+final readonly class Query_Block_Context implements Feature {
```

Comment:
As the CI results already noted, readonly classes aren't available while the plugin supports PHP 8.1.